rust/lance-datagen/src/generator.rs

```diff
@@ -1664,8 +1662,11 @@ pub mod array {
                 TimeUnit::Nanosecond => rand::<DurationNanosecondType>(),
             },
             DataType::Interval(unit) => match unit {
-                IntervalUnit::DayTime => rand::<IntervalDayTimeType>(),
-                IntervalUnit::MonthDayNano => rand::<IntervalMonthDayNanoType>(),
+                // TODO: fix these. In Arrow they changed to have specialized
+                // Native types, which don't support Distribution.
+                // IntervalUnit::DayTime => rand::<IntervalDayTimeType>(),
+                // IntervalUnit::MonthDayNano => rand::<IntervalMonthDayNanoType>(),
+                IntervalUnit::DayTime | IntervalUnit::MonthDayNano => todo!(),
```

Comment:
I added some utils to do this,
post it here if you would like to use it
https://github.com/lancedb/lance/pull/2589/files#diff-05a6cb2a51514a3883006392e5a3b6990e908be79d42414de787ba5c5b623c87R600

Reply:
Thanks!